cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -410,6 +418,11 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 	).Run()
 	time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
 
+	// run the shared informers
+	for _, informer := range informers {
+		go informer.Run(wait.NeverStop)
```

Comment:
We should have called the informer method that your register on
"AddMeAsLongAsIPromiseNotToMutateTheSharedCache"

On Thu, Apr 28, 2016 at 8:54 AM, David Eads <notifications@github.com>
wrote:

> In cmd/kube-controller-manager/app/controllermanager.go
> <https://github.com/kubernetes/kubernetes/pull/23575#discussion_r61421456>
> :
>
> > @@ -410,6 +418,11 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
> >  	).Run()
> >  	time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
> >
> > +	// run the shared informers
> > +	for _, informer := range informers {
> > +		go informer.Run(wait.NeverStop)
>
> Are shared informers ready for wider use, or are they still a work in
> progress?
>
> They are ready for wider use. The initial transitions of controllers to a
> shared pod informer went very smoothly, so I don't expect any structural
> problems. Remember not to mess with the object in the cache, its shared.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23575/files/f0c33d65b6fea84d791059d96c93a15e4b3de4ec#r61421456>
>